File: messaging/src/main/java/hu/bsstudio/robonaut/configuration/RabbitConfiguration.java
package hu.bsstudio.robonaut.configuration;

import org.springframework.amqp.core.Queue;
import org.springframework.amqp.rabbit.connection.ConnectionNameStrategy;
import org.springframework.amqp.rabbit.connection.SimplePropertyValueConnectionNameStrategy;
import org.springframework.amqp.support.converter.Jackson2JsonMessageConverter;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

@Configuration
public class RabbitConfiguration {

    @Bean
    public Queue generalTeamDataQueue() {
        return new Queue("general.teamData");
    }

    @Bean
    public Queue startSkillTimerQueue() {
        return new Queue("skill.timer");
    }

    @Bean
    public Queue skillRaceGateQueue() {
        return new Queue("skill.gate");
    }

    @Bean
    public Queue safetyCarFollowQueue() {
        return new Queue("speed.safetyCar.follow");
    }

    @Bean
    public Queue safetyCarOvertakeQueue() {
        return new Queue("speed.safetyCar.overtake");
    }

    @Bean
    public Queue startSpeedTimerQueue() {
        return new Queue("speed.timer");
    }

    @Bean
    public Queue speedRaceLapQueue() {
        return new Queue("speed.lap");
    }

    @Bean
    public Queue teamDataQueue() {
        return new Queue("team.teamData");
    }

    @Bean
    public Jackson2JsonMessageConverter jackson2JsonMessageConverter() {
        return new Jackson2JsonMessageConverter();
    }

    @Bean
    public ConnectionNameStrategy connectionNameStrategy() {
        return new SimplePropertyValueConnectionNameStrategy("spring.application.name");
    }
}

File: server/messaging/src/main/kotlin/hu/bsstudio/robonaut/configuration/RabbitConfiguration.kt
package hu.bsstudio.robonaut.configuration

import org.springframework.amqp.core.Queue
import org.springframework.amqp.rabbit.connection.ConnectionNameStrategy
import org.springframework.amqp.rabbit.connection.SimplePropertyValueConnectionNameStrategy
import org.springframework.amqp.support.converter.Jackson2JsonMessageConverter
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration

@Configuration
class RabbitConfiguration {
    @Bean
    fun generalTeamDataQueue() = Queue("general.teamData")

    @Bean
    fun startSkillTimerQueue() = Queue("skill.timer")

    @Bean
    fun skillRaceGateQueue() = Queue("skill.gate")

    @Bean
    fun safetyCarFollowQueue() = Queue("speed.safetyCar.follow")

    @Bean
    fun safetyCarOvertakeQueue() = Queue("speed.safetyCar.overtake")

    @Bean
    fun startSpeedTimerQueue() = Queue("speed.timer")

    @Bean
    fun speedRaceLapQueue() = Queue("speed.lap")

    @Bean
    fun teamDataQueue() = Queue("team.teamData")

    @Bean
    fun jackson2JsonMessageConverter() = Jackson2JsonMessageConverter()

    @Bean
    fun connectionNameStrategy(): ConnectionNameStrategy {
        return SimplePropertyValueConnectionNameStrategy("spring.application.name")
    }
}
